src/plugins/ui_feature/format.ts

```diff
@@ -48,18 +60,14 @@ export class FormatPlugin extends UIPlugin {
    * number value. Returns a default format if the cell hasn't format. Returns
    * undefined if no number value in the range.
    */
-  private searchNumberFormat(sheetId: UID, zones: Zone[]): Format | undefined {
-    for (let zone of zones) {
-      for (let row = zone.top; row <= zone.bottom; row++) {
-        for (let col = zone.left; col <= zone.right; col++) {
-          const cell = this.getters.getEvaluatedCell({ sheetId, col, row });
-          if (
-            cell.type === CellValueType.number &&
-            !cell.format?.match(DATETIME_FORMAT) // reject dates
-          ) {
-            return cell.format || createDefaultFormat(cell.value);
-          }
-        }
+  private getCellNumberFormat(positions: CellPosition[]): Format | undefined {
```

Comment:
unless I'm mistakne, you only call this function in 1 place and stricly with one position
```suggestion
  private getCellNumberFormat(position: CellPosition): Format | undefined {
```

Reply:
Update the code accordingly